### PR TITLE
Uicollection cell unit

### PR DESCRIPTION
**Pull request by kovalandrew**

### CODE REVIEW

File: sample/ios-app/src/Cells/SimpleCollectionCell.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//
+//  SimpleCollectionCell.swift
+//  TestProj
+//
+//  Created by Andrew Kovalev on 09.01.2020.
+//  Copyright © 2020 IceRock Development. All rights reserved.
+//
```
Comment (Alex009) [inline sample/ios-app/src/Cells/SimpleCollectionCell.swift]: replace to:
```swift
/*
 * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
 */
```

File: sample/ios-app/src/TestCollectionViewController.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        dataSource.units = units
+
+        self.dataSource = dataSource
+    }
+}
+
+extension TestCollectionViewController: TestingCollectionCollectionUnitFactory {
+    func createSimpleUnit(id: Int64, title: String, itemData: ItemData?) -> CollectionUnitItem {
+        // without R.swift
+//        return UITableViewCellUnit<SimpleCell>(
```
Comment (Alex009) [inline sample/ios-app/src/TestCollectionViewController.swift]: table?

File: sample/ios-app/src/TestCollectionViewController.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    
+    @IBOutlet private var collectionView: UICollectionView!
+    private var dataSource: UICollectionViewDataSource!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let testing = TestingCollection(unitFactory: self)
+
+        let dataSource = FlatUnitCollectionViewDataSource()
```
Comment (Alex009) [inline sample/ios-app/src/TestCollectionViewController.swift]: maybe use UnitCollectionViewDataSource?

Comment (Alex009) [reply sample/ios-app/src/TestCollectionViewController.swift]: https://github.com/icerockdev/moko-units/blob/40773b513d22094bf6f2ed3c7410e6d58c39b6f8/units/src/iosMain/kotlin/dev/icerock/moko/units/UnitCollectionViewDataSource.kt#L17

Comment (kovalandrew) [reply sample/ios-app/src/TestCollectionViewController.swift]: UnitTableViewDataSource and UnitCollectionViewDataSource are unavailable from Swift-native side

MultiPlatformLibrary.h

```
__attribute__((unavailable("Kotlin subclass of Objective-C class can't be imported")))
__attribute__((swift_name("UnitCollectionViewDataSource")))
@interface MPLUnitCollectionViewDataSource : NSObject
@end;

__attribute__((unavailable("Kotlin subclass of Objective-C class can't be imported")))
__attribute__((swift_name("UnitTableViewDataSource")))
@interface MPLUnitTableViewDataSource : NSObject
@end;

```

File: sample/ios-app/src/Cells/SimpleCollectionCell.swift (hunk 1 above)
Comment (kovalandrew) [reply sample/ios-app/src/Cells/SimpleCollectionCell.swift]: updated

File: sample/ios-app/src/TestCollectionViewController.swift (hunk 2 above)
Comment (kovalandrew) [reply sample/ios-app/src/TestCollectionViewController.swift]: Examples for "without Rswift" cases fixed for tableView and collectionView.

File: units/src/iosMain/swift/Core/CollectionView/Units/UICollectionViewCellUnit.swift [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import Foundation
```
Comment (Alex009) [inline units/src/iosMain/swift/Core/CollectionView/Units/UICollectionViewCellUnit.swift]: add copyright here 
```swift
/*
 * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
 */
```

File: units/src/iosMain/swift/Core/CollectionView/Units/UICollectionViewCellUnit.swift [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    guard let xibName: String = (Cell.self as? Reusable.Type)?.xibName?() ?? NSStringFromClass(type(of: Cell.self) as! AnyClass).components(separatedBy: ".").last else {
+      return
+    }
+    collectionView.register(UINib(nibName: xibName, bundle: Bundle.main), forCellWithReuseIdentifier: self.reuseId)
+	}
+
+  public typealias ConfiguratorType = ((_ cell: Cell) -> Void)
+
+  public var data: Cell.DataType //Собственно данные которые попадут в FillableProtocol.fill и .update
+  public var reuseId: String //обязательно указываем обычный reusable identifier
```
Comment (Alex009) [inline units/src/iosMain/swift/Core/CollectionView/Units/UICollectionViewCellUnit.swift]: translate please

File: sample/ios-app/src/Cells/SimpleCollectionCell.swift (hunk 1 above)
Comment (Alex009) [reply sample/ios-app/src/Cells/SimpleCollectionCell.swift]: not see changes

Comment (Alex009) [reply sample/ios-app/src/Cells/SimpleCollectionCell.swift]: my bad, all ok

File: units/src/iosMain/swift/Core/CollectionView/Units/UICollectionViewCellUnit.swift (hunk 4 above)
Comment (kovalandrew) [reply units/src/iosMain/swift/Core/CollectionView/Units/UICollectionViewCellUnit.swift]: Added

File: units/src/iosMain/swift/Core/CollectionView/Units/UICollectionViewCellUnit.swift (hunk 5 above)
Comment (kovalandrew) [reply units/src/iosMain/swift/Core/CollectionView/Units/UICollectionViewCellUnit.swift]: Translated

**Pull request merged**